modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -469,4 +472,16 @@ public static void closeAll(Collection<? extends AutoCloseable> closeables) thro
         if (ex != null)
             throw ex;
     }
+
+    /**
+     * Closes all provided objects.
+     *
+     * @param closeables Array of closeable objects to close.
+     * @throws Exception If failed to close.
+     *
+     * @see #closeAll(Collection)
+     */
+    public static void closeAll(AutoCloseable... closeables) throws Exception {
+        closeAll(Arrays.asList(closeables));
```

Comment:
why? Isn't this a bit of a premature optimization?

Reply:
@sashapolo It's just a good practice. But in this particular case we can ignore it. But in most cases (especially on hot path) it is best practice in Ignite to avoid redundant objects creation. We can't pray to escape analysis.